app/src/main/java/com/example/tasky/feature_agenda/presentation/AgendaViewModel.kt

```diff
@@ -14,99 +15,96 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.receiveAsFlow
 import kotlinx.coroutines.launch
-import java.text.SimpleDateFormat
-import java.util.Date
-import java.util.Locale
+import java.time.LocalDate
 import javax.inject.Inject
 
 @HiltViewModel
 class AgendaViewModel @Inject constructor(
     private val authenticatedRemoteRepository: AuthenticatedRemoteRepository,
-    private val sessionStateManager: SessionStateManager
+    private val sessionStateManager: SessionStateManager,
+    private val calendarHelper: CalendarHelper
 ) : ViewModel() {
 
     companion object {
         const val TAG = "AgendaViewModel"
     }
 
-    // viewState triggered by API response
-    private val _viewState =
-        MutableStateFlow<AgendaViewState?>(null)
-    val viewState: StateFlow<AgendaViewState?>
-        get() = _viewState
+    private val _viewState = MutableStateFlow(AgendaViewState())
+    val viewState: StateFlow<AgendaViewState> = _viewState
 
-    // viewEvent triggered by API response
     private val _viewEvent = Channel<AgendaViewEvent>()
     val viewEvent = _viewEvent.receiveAsFlow()
 
-    // UI changes via Composable
-    private val _showDialog = MutableStateFlow(false)
-    val showDialog: StateFlow<Boolean> get() = _showDialog
+    // UI does not change
+    val initials: String = sessionStateManager.getName()?.let { ProfileUtils.getInitials(it) } ?: ""
 
-    private val _initials = MutableStateFlow<String>("")
-    val initials: StateFlow<String> get() = _initials
-
-    private val _monthSelected = MutableStateFlow<String>(getCurrentMonth())
-    val monthSelected: StateFlow<String> get() = _monthSelected
-
-    private val _dateDialogState = MutableStateFlow<MaterialDialogState>(MaterialDialogState())
-    val dateDialogState: StateFlow<MaterialDialogState> get() = _dateDialogState
+    // not displayed on screen; only necessary for business logic
+    private val _yearSelected = MutableStateFlow(LocalDate.now().year)
+    val yearSelected: StateFlow<Int> get() = _yearSelected
```

Comment:
If it doesn't need to be accessed elsewhere, you can also leave it a single private property